packages/harmony/src/components/input/TextInput/TextInput.mdx

```diff
@@ -0,0 +1,240 @@
+import {
+  Canvas,
+  Meta,
+  Unstyled,
+  Title,
+  Subtitle,
+  Description,
+  Controls,
+  Primary
+} from '@storybook/blocks'
+
+import { Flex } from 'components'
+import { IconSearch } from 'icons'
+import { ComponentRules, RelatedComponents } from 'storybook/components'
+
+import { TextInput } from './TextInput'
+import * as TextInputStories from './TextInput.stories'
+import { TextInputSize } from './types'
+
+<Meta of={TextInputStories} />
+
+<Title />
+
+- [Overview](#overview)
+- [Props](#props)
+- [Configuration options](#configuration-options)
+- [Use cases and examples](#use-cases-and-examples)
+- [Do's and dont's](#dos-and-donts)
+- [Related Components](#related-components)
+
+## Overview
+
+<Subtitle />
+<Description />
+<Primary />
+
+## Props
+
+<Controls />
+
+## Configuration Options
+
+### Sizes
+
+<Canvas of={TextInputStories.Sizes} />
+
+### States
+
+<Canvas of={TextInputStories.DefaultStates} />
+<Canvas of={TextInputStories.SmallStates} />
+
+### Leading and trailing icons
+
+<Canvas of={TextInputStories.Icons} />
+
+### Assistive Text
+
+<Canvas of={TextInputStories.AssistiveText} />
+
+### Validation and counter
+
+Use validation to give feedback when an invalid input is provided. The validation error should appear when the user is done typing and out of focus.
+
+<Canvas of={TextInputStories.Validation} />
+
+### Special Inputs
+
+An input in which the trailing and leading icons are replaced with type (Title-large) and are positioned below the input’s label.
+
+<Canvas of={TextInputStories.SpecialInputs} />
+
+## Use cases and examples
+
+- Always use placeholder in input fields
```

Comment:
I know it's in the design, but a bit weird to have this twice (it's also a Do)

Reply:
yea it took me a couple of reads to figure out that it wasn't a mistake. not ideal for sure, but just matching the designs for now